### PR TITLE
Some missing ini stuff for bfme to pass unit test

### PR DESCRIPTION
**Pull request by Tarcontar**

### CODE REVIEW

File: src/OpenSage.Game/Data/Ini/IniDataContext.cs [hunk 1]
```diff
@@ -78,8 +93,18 @@ public sealed class IniDataContext
...
+        public Pathfinder Pathfinder { get; set; }
+
+        [AddedIn(SageGame.Bfme)]
+        public List<PlayerAIType> PlayerAITypes { get; } = new List<PlayerAIType>();
+
         public List<PlayerTemplate> PlayerTemplates { get; } = new List<PlayerTemplate>();
         public List<Rank> Ranks { get; } = new List<Rank>();
+
+        [AddedIn(SageGame.Bfme)]
+        public RegionCampain RegionCampain{ get; set; }
```
Comment (tgjones) [inline src/OpenSage.Game/Data/Ini/IniDataContext.cs]: Spotted a typo here, should be `RegionCampaign`.

Comment (Tarcontar) [reply src/OpenSage.Game/Data/Ini/IniDataContext.cs]: thats how its spelled inside the ini files, thought it would be easier to find in the code this way, but could rename it

File: src/OpenSage.Game.Tests/Data/Ini/IniFileTests.cs [hunk 2]
```diff
@@ -19,37 +19,45 @@ public IniFileTests(ITestOutputHelper output)
...
             {
                 GameDefinition.FromGame(SageGame.CncGenerals),
-                GameDefinition.FromGame(SageGame.CncGeneralsZeroHour)
+                GameDefinition.FromGame(SageGame.CncGeneralsZeroHour),
+                GameDefinition.FromGame(SageGame.Bfme),
+                //GameDefinition.FromGame(SageGame.Bfme2),
             };
 
-            InstalledFilesTestData.ReadFiles(".ini", _output, gameDefinitions, entry =>
+            foreach(var gameDefinition in gameDefinitions)
```
Comment (tgjones) [inline src/OpenSage.Game.Tests/Data/Ini/IniFileTests.cs]: Why this change?

File: src/OpenSage.Game/Data/Ini/WindowTransition.cs [hunk 3]
```diff
@@ -87,4 +124,62 @@ public enum WindowTransitionStyle
         [IniEnum("CONTROLBARARROW")]
         ControlBarArrow
     }
+
+    [AddedIn(SageGame.Bfme)]
+    public enum TransitionType
```
Comment (tgjones) [inline src/OpenSage.Game/Data/Ini/WindowTransition.cs]: Are you sure these are a different thing from WindowTransitionStyle?

File: src/OpenSage.Game.Tests/Data/Ini/IniFileTests.cs (hunk 2 above)
Comment (Tarcontar) [reply src/OpenSage.Game.Tests/Data/Ini/IniFileTests.cs]: this is to get the right game definition here:
var dataContext = new IniDataContext(entry.FileSystem, gameDefinition.Game);
the old approach was to use always SageGame.Generals, but since things like animations are game dependant this wouldnt work, so we need the proper game definition here.

File: src/OpenSage.Game/Data/Ini/WindowTransition.cs (hunk 3 above)
Comment (Tarcontar) [reply src/OpenSage.Game/Data/Ini/WindowTransition.cs]: True, this seems to be the same stuff, 
will fix it

**Comment by Tarcontar** [comment]
Combined the two enums
